### ISSUE TITLE
[Bug]: vllm-ascend v0.7.3.post1推理运行时，对于提示词中有JSON格式要求输出的情况会crash

### ISSUE THREAD

**Issue by kylewanginchina** [vllm-project/vllm-ascend#1592]
### Your current environment

环境信息：
OS: Kylin10
Ascend NPU Driver: Ascend-hdk-910b-npu-driver_23.0.7_linux-aarch64.run
Ascend NPU Firmware: Ascend-hdk-910b-npu-firmware_7.1.0.11.220.run
Ascend Docker Runtime: Ascend-docker-runtime_5.0.RC3.2_linux-aarch64.run
Docker: docker-ce-26.1.3-1.el8.aarch64.rpm
Containerd: containerd.io-1.6.32-3.1.el8.aarch64.rpm
vllm-ascend: vllm-ascend-v0.7.3.post1

32B模型推理：
export IMAGE=quay.io/ascend/vllm-ascend:v0.7.3.post1
docker run --rm
–name vllm-ascend-env
–device /dev/davinci0
–device /dev/davinci1
–device /dev/davinci2
–device /dev/davinci3
–device /dev/davinci_manager
–device /dev/devmm_svm
–device /dev/hisi_hdc
-v /usr/local/dcmi:/usr/local/dcmi
-v /usr/local/bin/npu-smi:/usr/local/bin/npu-smi
-v /usr/local/Ascend/driver/lib64/:/usr/local/Ascend/driver/lib64/
-v /usr/local/Ascend/driver/version.info:/usr/local/Ascend/driver/version.info
-v /etc/ascend_install.info:/etc/ascend_install.info
-v /root/.cache:/root/.cache
-p 8000:8000
-v /home/test:/mnt
-e VLLM_USE_MODELSCOPE=True
-e PYTORCH_NPU_ALLOC_CONF=max_split_size_mb:256
-it $IMAGE
vllm serve /mnt/Qwen3-32B --tensor-parallel-size 4 --max-model-len 16384 --gpu-memory-utilization 0.85

### 🐛 Describe the bug

一旦提示词中要求JSON输出，对应就会报错crash（正常测试情况不会），主要报错信息如下：

![Image](https://github.com/user-attachments/assets/cfd28845-0f89-4bf7-a323-dfb403918db0)

![Image](https://github.com/user-attachments/assets/203a46f8-3117-4ad7-8d43-38102422fbab)

测试了v0.7.3.post1原始版本，以及各个配置优化方式下的推理运行结果，对于JSON格式要求情况都会存在上述问题。而v0.8.5rc1以及v0.9.0rc2等没有上述问题，不过v0.8.5rc1没有计算图模式推理太慢，v0.9.0rc2有并发情况下会有bug（后面搜集完整新会提）也用不了。

**Comment by bazingazhou233-hub** [comment]
This issue has been resolved on the current `main` branch. The crash when requesting JSON output was caused by missing NPU support for structured output (xgrammar) in v0.7.3.post1.

**Root cause**: In v0.7.3.post1, vllm-ascend had no `apply_grammar_bitmask()` implementation for NPU. When a prompt requested JSON output, vLLM attempted to use the upstream GPU-based xgrammar path, which crashed on NPU with a device incompatibility error.

**Fix chain** (all merged to `main`):

1. **`1d88dacf` ([#475](https://github.com/vllm-project/vllm-ascend/pull/475))** — Added `supports_structured_output()` to the Ascend platform
2. **`65c1f457` ([#555](https://github.com/vllm-project/vllm-ascend/pull/555))** — Added NPU-specific `apply_grammar_bitmask()` that moves logits to CPU for bitmask application, then back to NPU
3. **`72eceff9` ([#2022](https://github.com/vllm-project/vllm-ascend/pull/2022))** — Fixed `IndexError` in the grammar bitmask method
4. **`e3c1ac89` ([#2524](https://github.com/vllm-project/vllm-ascend/pull/2524))** — Replaced custom implementation with upstream `apply_grammar_bitmask` (now NPU-compatible)

The current implementation in `model_runner_v1.py` correctly handles structured output on NPU:
```python
if grammar_output is not None:
    logits_dtype = logits.dtype
    logits = logits.to("cpu").float()
    apply_grammar_bitmask(scheduler_output, grammar_output, self.input_batch, logits)
    logits = logits.to(self.device).to(logits_dtype)
```

**Reproducibility**: Upgrade to vllm-ascend **v0.8.5rc1** or later. As you noted, this was already working in v0.8.5rc1+. JSON/structured output is fully functional on current `main`.
